aiida/storage/psql_dos/orm/nodes.py

```diff
@@ -312,21 +301,19 @@ class SqlaNodeCollection(BackendNodeCollection):
     ENTITY_CLASS: Type[SqlaNode] = SqlaNode
 
     def get(self, pk):
-        session = self.backend.get_session()
-
-        try:
-            return self.ENTITY_CLASS.from_dbmodel(
-                session.query(self.ENTITY_CLASS.MODEL_CLASS).filter_by(id=pk).one(), self.backend
-            )
-        except NoResultFound:
-            raise exceptions.NotExistent(f"Node with pk '{pk}' not found") from NoResultFound
+        with self.backend.transaction() as session:
+            try:
+                return self.ENTITY_CLASS.from_dbmodel(
+                    session.query(self.ENTITY_CLASS.MODEL_CLASS).filter_by(id=pk).one(), self.backend
+                )
+            except NoResultFound:
+                raise exceptions.NotExistent(f"Node with pk '{pk}' not found") from NoResultFound
 
     def delete(self, pk):
-        session = self.backend.get_session()
-
-        try:
-            row = session.query(self.ENTITY_CLASS.MODEL_CLASS).filter_by(id=pk).one()
-            session.delete(row)
-            session.commit()
-        except NoResultFound:
-            raise exceptions.NotExistent(f"Node with pk '{pk}' not found") from NoResultFound
+        with self.backend.transaction() as session:
+            try:
+                row = session.query(self.ENTITY_CLASS.MODEL_CLASS).filter_by(id=pk).one()
+                session.delete(row)
+                session.commit()
```

Comment:
doesn't the transaction context commit here anyway?

Reply:
removed it